### PR TITLE
Added interactors

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added interactors

### CODE REVIEW

File: interactors/blsKeysFetcher.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		if len(address.Hex) == 0 {
+			continue
+		}
+
+		blsKeys, err := fetcher.getBlsKeys(ctx, address, sender)
+		if err != nil {
+			return nil, err
+		}
+
+		timer.Reset(fetcher.timeBetweenQueries)
```
Comment (andreibancioiu) [inline interactors/blsKeysFetcher.go:77]: So that we don't stress the observers that much, correct?

File: interactors/blsKeysFetcher.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	requestBytes, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	responseBytes, statusCode, err := fetcher.httpClientWrapper.PostHTTP(ctx, endpoint, requestBytes)
+	if err != nil {
+		return nil, err
+	}
+	if !core.IsStatusCodeIs2xx(statusCode) {
```
Comment (andreibancioiu) [inline interactors/blsKeysFetcher.go:108]: This seems to be the old variant of the function (the old name) - has to be adjusted.

File: interactors/blsKeysFetcher.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+		return nil, fmt.Errorf("%w, but %d", errReturnCodeIsNotOk, statusCode)
+	}
+	response := &vmQueryResponse{}
+	err = json.Unmarshal(responseBytes, response)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make([]string, 0)
+	var accumulator []byte
```
Comment (andreibancioiu) [inline interactors/blsKeysFetcher.go:118]: Maybe rename `accumulator` to `returnedBlsKey` or maybe even do a loop with a step of 2 to walk over the resulted data? (optional, can also stay as it is).

File: interactors/validatorStatistics.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+// Query will query the validator statistics API endpoint route
+func (interactor *validatorStatisticsInteractor) Query(ctx context.Context) (map[string]*core.ValidatorStatistics, error) {
+	log.Debug("validatorStatisticsInteractor.Query")
+	bytes, statusCode, err := interactor.httpClientWrapper.GetHTTP(ctx, validatorStatisticsURL)
+	if err != nil {
+		return nil, err
+	}
+	if !core.IsStatusCodeIs2xx(statusCode) {
```
Comment (andreibancioiu) [inline interactors/validatorStatistics.go:41]: Old function name, should be updated.

File: interactors/validatorStatistics_test.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+				return []byte(`not-a-json-response`), http.StatusOK, nil
+			},
+		}
+		instance, _ := NewValidatorStatisticsInteractor(wrapper)
+		response, err := instance.Query(context.Background())
+		assert.Nil(t, response)
+		assert.NotNil(t, err)
+		assert.Contains(t, err.Error(), "invalid character")
+	})
+	t.Run("http client wrapper return an invalid response should return error", func(t *testing.T) {
```
Comment (andreibancioiu) [inline interactors/validatorStatistics_test.go:89]: Typo / returns.

File: interactors/blsKeysFetcher.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/multiversx/mx-chain-keys-monitor-go/core"
+
```
Comment (miiu96) [inline interactors/blsKeysFetcher.go:11]: remove empty line

File: interactors/blsKeysFetcher.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/multiversx/mx-chain-core-go/core/check"
+)
+
+const getBlsKeysStatusFuncName = "getBlsKeysStatus"
```
Comment (miiu96) [inline interactors/blsKeysFetcher.go:15]: maybe you can group constants togheder

File: interactors/blsKeysFetcher_test.go [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-keys-monitor-go/core"
+	"github.com/multiversx/mx-chain-keys-monitor-go/mock"
+
```
Comment (miiu96) [inline interactors/blsKeysFetcher_test.go:15]: remove empty line

File: interactors/validatorStatistics.go [hunk 9]
```diff
@@ -0,0 +1,62 @@
+package interactors
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/multiversx/mx-chain-keys-monitor-go/core"
+
```
Comment (miiu96) [inline interactors/validatorStatistics.go:9]: remove empty line

File: interactors/validatorStatistics_test.go [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+import (
+	"context"
+	"errors"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/multiversx/mx-chain-keys-monitor-go/core"
+	"github.com/multiversx/mx-chain-keys-monitor-go/mock"
+
```
Comment (miiu96) [inline interactors/validatorStatistics_test.go:12]: remove empty line

File: interactors/blsKeysFetcher.go (hunk 1 above)
Comment (iulianpascalau) [reply interactors/blsKeysFetcher.go:77]: yes. Usually 5 minutes is enough

File: interactors/blsKeysFetcher.go (hunk 2 above)
Comment (iulianpascalau) [reply interactors/blsKeysFetcher.go:108]: Of course. Fixed.

File: interactors/blsKeysFetcher.go (hunk 3 above)
Comment (iulianpascalau) [reply interactors/blsKeysFetcher.go:118]: renamed to `returnedBlsKey`

Comment (iulianpascalau) [reply interactors/blsKeysFetcher.go:118]: also refactored the function

File: interactors/validatorStatistics.go (hunk 4 above)
Comment (iulianpascalau) [reply interactors/validatorStatistics.go:41]: updated

File: interactors/validatorStatistics_test.go (hunk 5 above)
Comment (iulianpascalau) [reply interactors/validatorStatistics_test.go:89]: fixed

File: interactors/blsKeysFetcher.go (hunk 6 above)
Comment (iulianpascalau) [reply interactors/blsKeysFetcher.go:11]: done

File: interactors/blsKeysFetcher.go (hunk 7 above)
Comment (iulianpascalau) [reply interactors/blsKeysFetcher.go:15]: indeed, done

File: interactors/blsKeysFetcher_test.go (hunk 8 above)
Comment (iulianpascalau) [reply interactors/blsKeysFetcher_test.go:15]: done

File: interactors/validatorStatistics.go (hunk 9 above)
Comment (iulianpascalau) [reply interactors/validatorStatistics.go:9]: done

File: interactors/validatorStatistics_test.go (hunk 10 above)
Comment (iulianpascalau) [reply interactors/validatorStatistics_test.go:12]: done

**Review by andreibancioiu — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**